azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -277,19 +277,17 @@ def _execute(
             request_headers["Authorization"] = self._aad_helper.acquire_authorization_header()
 
         # trace http post call for response
-        http_trace_attributes = KustoTracingAttributes.create_http_attributes(url=endpoint, method="POST", headers=request_headers)
-        response = KustoTracing.call_func_tracing(
-            self._session.post,
+        invoker = lambda: self._session.post(
             endpoint,
             headers=request_headers,
             json=json_payload,
             data=payload,
             timeout=timeout.seconds,
             stream=stream_response,
             allow_redirects=False,
-            name_of_span="KustoClient.http_post",
-            tracing_attributes=http_trace_attributes,
         )
+        http_trace_attributes = SpanAttributes.create_http_attributes(url=endpoint, method="POST", headers=request_headers)
+        response = Span.run(invoker, name_of_span="KustoClient.http_post", tracing_attributes=http_trace_attributes)
```

Comment:
Here I think you should break it to lines and inline http_trace_attributes (you don't need the param names also)

Reply:
done - removed params from http_trace but left params for MonitoredActivity.invoke

Reply:
Note that in some cases black. reformats so that it is on the same line